external-crates/move/crates/move-package-alt/src/errors.rs

```diff
@@ -4,14 +4,229 @@
 
 use std::{ops::Range, path::PathBuf};
 
+use codespan_reporting::{
+    diagnostic::{Diagnostic, Label},
+    files::SimpleFiles,
+    term::{
+        self,
+        termcolor::{ColorChoice, StandardStream},
+    },
+};
 use serde::{Deserialize, Serialize};
 use serde_spanned::Spanned;
+use thiserror::Error;
 
-// TODO: sensible error type
-pub type PackageError = anyhow::Error;
+/// The main error type for package-related operations
+#[derive(Error, Debug)]
+pub enum PackageError {
+    #[error(transparent)]
+    Manifest(#[from] ManifestError),
 
-// TODO: sensible error type
-pub type PackageResult<T> = anyhow::Result<T>;
+    #[error(transparent)]
+    Io(#[from] std::io::Error),
+
+    #[error(transparent)]
+    Other(#[from] anyhow::Error),
+
+    #[error(transparent)]
+    Toml(#[from] toml_edit::de::Error),
+
+    #[error(transparent)]
+    Codespan(#[from] codespan_reporting::files::Error),
+}
+
+/// Result type for package operations
+pub type PackageResult<T> = Result<T, PackageError>;
+
+/// Errors that can occur during manifest parsing
+#[derive(Error, Debug)]
+pub enum ManifestError {
+    /// Invalid package name
+    #[error("Invalid package name: {message}")]
+    InvalidPackageName {
+        message: String,
+        span: Option<Range<usize>>,
+        path: PathBuf,
+        source_code: String,
+    },
+
+    /// Invalid edition
+    #[error("Invalid edition: {message}")]
+    InvalidEdition {
+        message: String,
```

Comment:
I also think having a separate message somewhat defeats the point of this_error. I would expect the message to come entirely from the `#[error(...)]` annotation. So here, for example, I might have `InvalidEdition` contain the edition, rather than the message.